### PR TITLE
Update app server versions

### PR DESCRIPTION
**Pull request by littleaj**
updates smoke test containers to latest version of their respective app servers.

also, adds readmes to each app server directory for any special instructions.

### CODE REVIEW

File: test/smoke/appServers/JBossEAP.6/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# JBoss EAP 6
+
+[Supported JRE Versions](https://access.redhat.com/articles/111663): <= Java 8
+
+Set `JBOSSEAP6_ZIPFILE` to point to the zip file containing JBoss EAP 6.
+
+This is configured to run on Java 7 & 8, so a version >= 6.3.3 is needed; 6.4.0 is reccommended.
```
Comment (timja) [inline test/smoke/appServers/JBossEAP.6/README.md]:
```suggestion
This is configured to run on Java 7 & 8, so a version >= 6.3.3 is needed; 6.4.0 is recommended.
```

Comment (littleaj) [reply test/smoke/appServers/JBossEAP.6/README.md]: good catch, thanks :)

**Comment by timja** [comment]
anything holding this up @littleaj or @trask ?

**Comment by trask** [comment]
Oh, sorry, I missed this.  Approved!

**Comment by littleaj** [comment]
Just running the tests before merge. tomcat changed to sha512sum verification, so I updated that. Also, tomcat 8 is end-of-life so I removed that from the list. tomcat 7 & 8.5 are still being tested.

**Comment by littleaj** [comment]
the build is passing; the github integration just doesn't know the difference between failed and cancelled. smoke tests seem to be working after merging in master. I'll merge this when it finishes.

**Pull request merged**